src/frontend/src/common/components/library/data_subview/components/EditSamplesConfirmationModal/components/ImportFile/index.tsx

```diff
@@ -167,13 +172,34 @@ export default function ImportFile({
             | NamedGisaidLocation
             | undefined) = passOrDeleteEntry(value);
         }
+        const filledInUploadedMetadata = {
+          ...pick(uploadedMetadataEntry, uploadedFieldsWithData),
+        };
+
+        if (!isEmpty(uploadedMetadataEntry)) {
+          // check if there is any missing data that the user needs to fill in before proceeding
+          setMissingData((prevMissingData) => {
+            const rowMissingMetadataWarnings = warnMissingMetadata({
+              ...existingMetadataEntry,
+              ...filledInUploadedMetadata,
+            });
+            if (rowMissingMetadataWarnings) {
+              return {
+                ...prevMissingData,
+                [sampleId]: rowMissingMetadataWarnings,
+              };
+            } else {
+              return { ...prevMissingData };
```

Comment:
wouldn't this just be `return prevMissingData`? Or is there a reason to spread here?

Reply:
I agree, this doesn't seem necessary, but maybe there's some reason to want it to be a pure function? I know that's a big thing with redux, I don't remember if there's a reason to follow that pattern with React state though.

Reply:
removed